### DISCUSSION TITLE
New Running And Debugging System for Android And Ios &gui apps

### DISCUSSION THREAD

**Discussion by Mohamedjcali** [godotengine/godot-proposals#8627] [GUI]
**Describe the project you are working on**
iam working app (not mobile game) normal books app firstly tried to use react native Expo because my computer cant able to run the normal emulator especially android studio virtual device then i remembered Godot i am not provisional game developer but was using Godot like 1 year settled up Godot for Mobile Setup

Describe the problem or limitation you are having in your project
Godot was especially designed for games that seems weak point there is more app developers then is game developers also Godot is lightweight and much faster in any mobile app develop tool even java but the problem is if you need it to develop app you have to run it mobile a lot and Godot when ever you need to run app in android what is doing is downloading the whole app and run it like litterly is downloading the whole app and if you need to run another time is must downloaded the whole app yes godot sometimes is big like if you using 3d and yeah alot game devlopers is need that

**Describe the feature / enhancement and how it helps to overcome the problem or limitation**
godot version that was used to devlop gui apps that runs in every system especial mobile and desktop this time we are not becoming like unity competitor but flutter react native competitor and have expo run system

**Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams**
what we can do is create new godot version i mean it like godot -m v.1.0.0 that is based on godot but this time the focus is not to create powerfull tool but usefull tool like remove the 3d section create run sytem that is like expo react native system that every person can able to run it also design simple api system that non opensource systems can use techanicaly become like flutter but we using gd script buitfull editor section opject system also add back end system and even made gd script server language.

If this enhancement will not be used often, can it be worked around with a few lines of script?
this is using and based godot main like if new seystem is out we are using that if new node is add especialy if that node is 2d also it making much faster the system like we are not using any 3d sysytem

Is there a reason why this should be core and not an add-on in the asset library?
this improving what does the people use and not only game developers using Godot

**Comment by hsandt** [comment]
I see a duplicate in https://github.com/godotengine/godot-proposals/discussions/8626, which one do you want to keep? This one in GUI or the other one in Editor?
